java/src/jmri/jmrit/logix/WarrantTableAction.java

```diff
@@ -201,6 +202,31 @@ synchronized protected static void writetoLog(String text) {
         }
     }
 
+    synchronized protected static void closeNXFrame(NXFrame frame) {
+        if (frame != null) {
+            if (frame.equals(_nxFrame)) {
+                _nxFrame = null;
+            }
+            frame.dispose();
+        }
+    }
+
+    synchronized protected static boolean setNXFrame(NXFrame frame) {
+        if (_nxFrame !=null && _nxFrame != frame) {
+            frame.toFront();
+            if (JOptionPane.YES_OPTION == JOptionPane.showConfirmDialog(_nxFrame, Bundle.getMessage("NXWarrantInProgress"),
+                    Bundle.getMessage("WarningTitle"), JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE) )
+                return false;
+        }
+        closeNXFrame(_nxFrame);
```

Comment:
Not sure of the logic here, my apologies if I'm wrong, but should the `closeNXFrame(_nxFrame);` call be inside the test for a different NXFrame already existing?  As it it now, it you call `setNXFrame` twice with the same `frame1 value, it looks like it'll `dispose()` the NXFrame that you're currently using.